File: src/main/java/diff/patch/accessors/MapGetter.java
package diff.patch.accessors;

import diff.patch.Getter;

import java.util.Map;

/**
 *
 * @author sebasjm <smarchano@primary.com.ar>
 */
public class MapGetter<Key, Value> implements Getter<Value> {
    final Map<Key, Value> target;
    final Key field;

    public MapGetter(Map<Key, Value> target, Key field) {
        this.target = target;
        this.field = field;
    }

    @Override
    public Value get() {
        return target.get(field);
    }
    
}

File: src/main/kotlin/diff/patch/accessors/MapGetter.kt
package diff.patch.accessors

import diff.patch.Getter

/**

 * @author sebasjm <smarchano></smarchano>@primary.com.ar>
 */
class MapGetter<Key, Value: Any>(internal val target: Map<Key, Value>, internal val field: Key) : Getter<Value> {

    override fun get(): Value? {
        return target[field]
    }

}
